### ISSUE TITLE
git ship: does not work when merge queues are enabled on Github

### ISSUE THREAD

**Issue by breml** [git-town/git-town#3294]
### Steps to reproduce

1. Have a repository, where [merge queues](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-a-merge-queue) are required (repository needs to be owned by an organization, details in the previous link).
2. Have a PR ready to be shipped.
3. Execute `git town ship`

### Command output

```shell
$ git town ship

[branch] git fetch --prune --tags
Enter passphrase for key '.../id_ed25519_sk': 

[branch] git checkout master
Switched to branch 'master'
Your branch is up to date with 'origin/master'.

[master] git merge --squash branch
Updating b0fce7528..980483e8a
Fast-forward
Squash commit -- not updating HEAD
<redacted>
 4 files changed, 4 insertions(+), 4 deletions(-)

[master] git commit
[master be9366e8b] commit message
 4 files changed, 4 insertions(+), 4 deletions(-)

[master] git reset --hard HEAD~1
HEAD is now at b0fce7528 some other commit message

GitHub API: merging PR #5216 ... ok

Error: PUT https://api.github.com/repos/org/repo/pulls/5216/merge: 405 Changes must be made through the merge queue []
Auto-undo... 


[master] git checkout branch
Switched to branch 'branch'
Your branch is up to date with 'origin/branch'.

Error: PUT https://api.github.com/repos/org/repo/pulls/5216/merge: 405 Changes must be made through the merge queue []
```
```


### Additional information

_No response_

**Comment by kevgo** [comment]
Some things that Git Town could do here:

- [ ] display a proper error message
- [ ] provide a config setting for `.git-branches.toml` to disable ship for this repo since it uses a merge queue and ship doesn't work

**Comment by breml** [comment]
Aren't merge queues available through the github api as well? I would have hoped, that Git Town could have a config setting that does trigger merge queue instead of a regular merge/squash of a PR.

**Comment by kevgo** [comment]
Good point. I forgot that GitHub's merge queue really only replaces the "merge" button and doesn't do the testing. `git ship` should detect the merge queue and merge the PR through it instead of merging directly.

Having said this, most people probably don't need to use `git ship` at all. This command was invented before GitHub was in widespread use. Nowadays, it's often easier to use Git Town to create and sync branches, and then review and merge them (with or without the merge queue) using the GitHub UI.

**Comment by breml** [comment]
Thanks for your explanation about the use of `git town ship`. For me, the complete workflow with Git Town in combination with Github is not yet full clear, especially when a PR is ready to be merged and then the cleanup should happen. What do you advise in this case?

For me, the workflow should look something like this:

1. `git town hack feature-1`
2. work on `feature-1` as normal (`git add`, `git commit`, `git push`)
3. create PR on GH, wait for approval (should I use `git town propose` for this?)
4. start work on dependent feature 2: `git town append feature-2`
5. work on `feature-2` as normal (`git add`, `git commit`, `git push`)
7. Received some feedback for `feature-1`, `git town switch`, `git town sync`, `git add`, `git commit`, `git push`
8. Continue on `feature-2`: `git town switch`, `git town sync` (add, commit, push)
9. create PR on GH, wait for approval (should I use `git town propose` for this?)
10. `feature-1` is approved, now what? merge on GH or use `git town ship`?
  after the merge, how do I properly cleanup `feature-1` and make sure that `feature-2` now has `master` as parent and is sync properly?

Thanks for your explanation.

**Comment by breml** [comment]
If I get your answer in https://github.com/git-town/git-town/issues/3298#issuecomment-2038348678 correctly, I just run `git town sync` in step 10 of the above message and everything should be cleaned up "automagically", correct?

**Comment by kevgo** [comment]
Your questions are legit. I'm planning a series of screencasts that explain this better in the coming weeks.

> create PR on GH, wait for approval (should I use git town propose for this?)

You can use `git town propose` for this but don't have to. All it does is open the web page to create the PR for you.

> 9. feature-1 is approved, now what? merge on GH or use git town ship?

You can do either. If you use `git town ship`, it also updates the PRs of all child branches. That requires support for the merge queue though.

Most people merge through the GitHub UI these days. I think GitHub is now smart enough to update PRs of child branches, but I need to verify that (am traveling right now). Then run `git sync` to get your local workspace up to date.

**Comment by kevgo** [comment]
@stephenwade You are right that the actual commits in the branches don't get updated. You can update them by running `git town sync --all`.

I meant that GitHub updates the _metadata_ of pull requests for child branches when shipping a branch: it changes the head branch from the now shipped branch to the parent of the now shipped branch.

So my recommendation for the workflow that @breml describes is:

1. `git town hack feature-1`
2. work on feature-1 as normal (git add, git commit, git push)
3. run `git town propose` and create PR for branch feature-1 on GH
4. start work on dependent feature 2: `git town append feature-2`
5. work on feature-2 as normal (git add, git commit, git push)
6. received some feedback for feature-1: `git town switch`, `git town sync`, git add, git commit, git push
7. continue on feature-2: `git town switch`, `git town sync`, add, commit, push
8. run `git town propose` to create a PR for branch `feature-2` on GH, wait for approval
9. `feature-1` is approved: merge it by clicking the "merge" button in the GitHub UI (or whatever other button merges the branch there)
10. note how GitHub has updated the parent branch for the PR of feature-2 from "feature-1" to "main"
11. run `git town sync --all` to cleanup the local `feature-1` branch, sync the commits in your local `main` branch, and sync the commits in the (local and remote) `feature-2` branch
